### DISCUSSION TITLE
Fastlane Support

### DISCUSSION THREAD

**Discussion by jay8t6** [airsdk/Adobe-Runtime-Support#914] [Ideas]
It would be nice if we could automate the AIR app build process via fastlane cli. Would it be possible to create the builds from the CLI so we can integrate it with fastlane and automate the entire flow?

**Comment by leossmith** [comment]
I believe that with the new beta version using gradle to build app bundles, this is easier to do.

**Reply by ajwfrost** [reply]
Yes, we'll have to see how that works - building everything via a command-line or via ant or whatever, should all be possible but it looks like the fastlane folk would be more looking at using Gradle as well..  so it might be a good fit with the 33.1.1.533 release, we have a flag to allow the Android Studio-compatible project files to be kept around, although we may need to refine this to get it working with these sorts of tools..

**Reply by jay8t6** [reply]
Thats great to hear @ajwfrost! Keep us updated on the progress.

**Reply by dgurney75** [reply]
Will it work even if we build the app in Animate CC? Would love to have a deployment team package and send the app to the app stores without having them install Creative Cloud just to get Animate CC for publishing.

**Reply by ajwfrost** [reply]
If you're building with Animate, then there are two steps to the packaging process:
a) Creating the SWF file - which, if you have a .fla file and any library assets etc, you will need Animate for
b) Packaging the SWF and other assets into an IPA/APK/AAB - which is what you use ADT for, or if you're using Animate, then Animate calls ADT for you.

But we can't do the creation of SWF files from the graphical assets that are in a .fla file. The AIR SDK comes with an ActionScript compiler but not with anything that can cope with the .fla format...

So .. if you're not actually changing your SWFs, then it would be possible to package up and deploy the application using the AIR SDK without needing Animate i.e. if it's just to cope with new rules from the app stores.

**Reply by leossmith** [reply]
> Yes, we'll have to see how that works - building everything via a command-line or via ant or whatever, should all be possible but it looks like the fastlane folk would be more looking at using Gradle as well..  so it might be a good fit with the 33.1.1.533 release, we have a flag to allow the Android Studio-compatible project files to be kept around, although we may need to refine this to get it working with these sorts of tools..
> 

This sounds like a very good turn. Being able to change parts of the android project in android studio is a very important part of the process. Maybe @ajwfrost  can check the export process of flutter to get ideas for that. For example, adaptive icons are created directly in android studio generated project.
